scripts/upgrade-hooks/1.1.0.ts

```diff
@@ -0,0 +1,251 @@
+    throw new Error("No data found in GraphQL response");
+  }
+
+  return result.data.offers;
+}
+
+/**
+ * Calculate fees for backfilling based on offer details.
+ */
+function calculateFees(verifierFee: bigint, facilitatorFeePercentBps: bigint, tokenId: string, price: bigint): FeeData {
+  const HUNDRED_PERCENT_BPS = 100_00n;
+  const BOSON_PROTOCOL_FEE_BPS = 50n; // 0.5%
+  const bosonProtocolFee = (price * BOSON_PROTOCOL_FEE_BPS) / HUNDRED_PERCENT_BPS;
+  const fermionFeeAmount = 0n;
+  const remainder = price - bosonProtocolFee - fermionFeeAmount - verifierFee;
+  const facilitatorFeeAmount = remainder > 0n ? (remainder * facilitatorFeePercentBps) / HUNDRED_PERCENT_BPS : 0n;
+
+  return {
+    tokenId,
+    bosonProtocolFee,
+    fermionFeeAmount,
+    verifierFee,
+    facilitatorFeeAmount,
+  };
+}
+
+/**
+ * Prepare backfill data from fetched offers.
+ */
+export async function prepareFeeBackfillData(client: any): Promise<FeeData[]> {
+  const offers = await fetchGraphQLData(client);
+  const feeDataList: FeeData[] = [];
+
+  for (const offer of offers) {
+    const verifierFee = BigInt(offer.verifierFee);
+    const facilitatorFeePercentBps = BigInt(offer.facilitatorFeePercent);
+
+    for (const token of offer.fNFTs) {
+      const price = token.priceLog?.amount ? BigInt(token.priceLog.amount) : 0n;
+      const tokenId = token.id.split("-")[1];
+      const feeData = calculateFees(verifierFee, facilitatorFeePercentBps, tokenId, price);
+      feeDataList.push(feeData);
+    }
+  }
+
+  console.log(
+    "Prepared Backfill Data:",
+    JSON.stringify(feeDataList, (_key, value) => (typeof value === "bigint" ? value.toString() : value), 2),
+  );
+
+  return feeDataList;
+}
+
+/**
+ * Perform pre-upgrade tasks, including deploying the BackfillingV1_1_0 contract,
+ * preparing initialization data, and making the diamond cut.
+ */
+export async function preUpgrade(protocolAddress: string, chainId: number, env: string) {
+  // Set the correct GRAPHQL_URL based on chainId and env
+  const graphQLUrl = getGraphQLUrl(chainId, env);
+  const client = createClient({
+    url: graphQLUrl,
+    exchanges: [fetchExchange],
+    fetch: fetch as any,
+  });
+
+  console.log("Fetching and preparing backfill data...");
+  const feeDataList = await prepareFeeBackfillData(client);
+  const offerDataList = await prepareOfferBackfillData(client);
+
+  console.log("Deploying BackfillingV1_1_0...");
+  const BackfillingV1_1_0 = await ethers.getContractFactory("BackfillingV1_1_0");
+  const backfillingFacet = await BackfillingV1_1_0.deploy();
+  await backfillingFacet.deployed();
+  console.log(`BackfillingV1_1_0 deployed at: ${backfillingFacet.address}`);
+
+  console.log("Preparing initialization calldata...");
+  const backFillFeesCalldata = backfillingFacet.interface.encodeFunctionData("backFillTokenFees", [feeDataList]);
+  const backFillOfferCalldata = backfillingFacet.interface.encodeFunctionData("backFillOfferData", [offerDataList]);
```

Comment:
Maybe it's worth doing an estimate how much data can we fill in in a single tx before running into gasLimit. If there is too much data, we might need to split it into multiple calls. 

But that's not a scope of this PR, since I don't really expect a problem (at least not on mainnets).